### PR TITLE
bpo-47129: Add more informative messages to f-string syntax errors

### PR DESCRIPTION
**Pull request by macgors**
https://bugs.python.org/issue47129

Automerge-Triggered-By: GH:lysnikolaou

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@macgors

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


We also demand... [A SHRUBBERY!](https://www.youtube.com/watch?v=zIV4poUZAQo)


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by lysnikolaou — APPROVED** [review]
Thanks @macgors for the PR. LGTM! Please sign the CLA, so we can merge it.

**Comment by macgors** [comment]
> Thanks @macgors for the PR. LGTM! Please sign the CLA, so we can merge it.

Already signed it, waiting for it to be review/accepted, should come within 1 US business day.

File: Parser/string_parser.c [hunk 1]
```diff
@@ -357,9 +357,15 @@ fstring_compile_expr(Parser *p, const char *expr_start, const char *expr_end,
...
         }
     }
+    
     if (s == expr_end) {
-        RAISE_SYNTAX_ERROR("f-string: empty expression not allowed");
-        return NULL;
+        if (*expr_end == '!' || *expr_end == ':' || *expr_end == '=') {
+            RAISE_SYNTAX_ERROR("f-string: expression required before '%c'", *expr_end);
+            return NULL;
+        } else {
```
Comment (ericvsmith) [inline Parser/string_parser.c]: I know this is a nit, but could you remove the else clause, and move the next two statements out one level? When reviewing code it's more obvious that all code paths have the "return NULL" in them.

**Review by ericvsmith — CHANGES_REQUESTED** [review]
Otherwise, this looks ready to merge. Thanks!

**Comment by macgors** [comment]
No problem, @ericvsmith. I have made the requested changes; please review again.

**Review by ericvsmith — APPROVED** [review]

**Review by lysnikolaou — APPROVED** [review]

**Pull request merged**